File: scripts/graphcraft/assets/packages/ui-core/flutter/lib/components/button_primary.dart
import 'package:flutter/material.dart';

import '../tokens.dart';

/// Primary CTA — max one per screen (design-system/components/button.example.yaml)
/// @graphcraft component:button-primary
class ButtonPrimary extends StatelessWidget {
  const ButtonPrimary({
    super.key,
    required this.label,
    this.onPressed,
    this.loading = false,
  });

  final String label;
  final VoidCallback? onPressed;
  final bool loading;

  @override
  Widget build(BuildContext context) {
    return SizedBox(
      width: double.infinity,
      child: ElevatedButton(
        onPressed: loading ? null : onPressed,
        style: ElevatedButton.styleFrom(
          backgroundColor: DesignTokens.colorActionPrimary,
          foregroundColor: DesignTokens.colorTextPrimary,
          minimumSize: Size(0, DesignTokens.touchTargetMin),
          padding: EdgeInsets.symmetric(
            horizontal: DesignTokens.spacingButtonPadding * 2,
            vertical: DesignTokens.spacingButtonPadding,
          ),
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(DesignTokens.radiusDefault),
          ),
        ),
        child: loading
            ? SizedBox(
                height: 20,
                width: 20,
                child: CircularProgressIndicator(
                  strokeWidth: 2,
                  color: DesignTokens.colorTextPrimary,
                ),
              )
            : Text(label, style: const TextStyle(fontWeight: FontWeight.w600)),
      ),
    );
  }
}

File: packages/ui-core/unity/Runtime/Components/ButtonPrimary.cs
using UnityEngine;
using UnityEngine.UI;

namespace GraphCraft.UI
{
    /// <summary>
    /// Primary CTA — design-system/components/button.example.yaml
    /// @graphcraft component:button-primary
    /// </summary>
    [RequireComponent(typeof(Button))]
    [RequireComponent(typeof(LayoutElement))]
    public class ButtonPrimary : MonoBehaviour
    {
        public Text labelText;
        public Button button;

        void Reset()
        {
            button = GetComponent<Button>();
            var le = GetComponent<LayoutElement>();
            le.minHeight = DesignTokens.TouchTargetMin;

            if (ColorUtility.TryParseHtmlString(DesignTokens.ColorActionPrimary, out var primary))
            {
                var colors = button.colors;
                colors.normalColor = primary;
                button.colors = colors;
            }

            var image = GetComponent<Image>();
            if (image != null && ColorUtility.TryParseHtmlString(DesignTokens.ColorActionPrimary, out var bg))
            {
                image.color = bg;
            }
        }

        public void SetLabel(string text)
        {
            if (labelText != null)
            {
                labelText.text = text;
            }
        }
    }
}
